### PR TITLE
fix(node): only count service as endpoints when they are global

### PR DESCRIPTION
**Pull request by hydrogen18**
Changes 

1. Only count as an endpoint whenever the "expose" is global
2. HTTP services (kube ingress) need at least one host - this avoids an error in kubernetes
3. If there is no "to" in SDL still define a kube service
4. Put HTTP services in ClusterIP service in kube, not NodePort

### CODE REVIEW

**Review by boz — APPROVED** [review]

**Review by Ropes — APPROVED** [review]

**Comment by hydrogen18** [comment]
This worked with this SDL 

```
---
version: "2.0"
services:
  redis:
    image: redis:6.0.9-buster
    expose:
      - port: 6379
        as: 6379

  retwis:
    image: fager/retwis
    expose:
      - port: 8080
        as: 80
        accept:
          - retwis.localhost
        to:
          - global: true
    env:
      - REDIS_OPTIONS_HOST=redis-np
    dependencies:
      - service: redis
profiles:
  compute:
    retwis:
      resources:
        cpu:
          units: 0.5
        memory:
          size: 128Mi
        storage:
          size: 256Mi
    redis:
      resources:
        cpu:
          units: 0.5
        memory:
          size: 512Mi
        storage:
          size: 512Mi
  placement:
    westcoast:
      pricing:
        retwis: 
          denom: uakt
          amount: 9000
        redis:
          denom: uakt
          amount: 9000
deployment:
  retwis:
    westcoast:
      profile: retwis
      count: 1
  redis:
    westcoast:
      profile: redis
      count: 1
```

It created both services

services


And one of the services has an ingress pointed at it since it is port 80 && global == true

ingress

File: provider/cluster/util/util.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package util
 
 import "github.com/ovrclk/akash/manifest"
 
-func ShouldExpose(expose manifest.ServiceExpose) bool {
+func ShouldBeIngress(expose manifest.ServiceExpose) bool {
```
Comment (boz) [inline provider/cluster/util/util.go:5]: :pray:

File: validation/manifest.go [hunk 2]
```diff
@@ -103,6 +103,10 @@ func validateServiceExpose(serviceName string, serviceExpose manifest.ServiceExp
 		helper.globalServiceCount++
 	}
 
+	if util.ShouldBeIngress(serviceExpose) && 0 == len(serviceExpose.Hosts) {
```
Comment (boz) [inline validation/manifest.go:106]: `accept:` is optional - a unique hostname will be generated for the service.

**Review by boz — COMMENTED** [review]
Looks great, but we don't require a hostname in `accept:`.

**Comment by boz** [comment]
superseded by #952

**Pull request closed without merging**